File: mizhara-web/src/types/offer.ts
export type OfferType = "percentage" | "fixed" | "bogo";
export type OfferScope = "all" | "selected";

export type Offer = {
  id: string;
  name: string;
  description: string;
  image?: string;
  type: OfferType;
  scope: OfferScope;
  percentage?: number;
  fixedAmount?: number;
  minPurchase?: number;
  maxDiscount?: number;
  buyQuantity?: number;
  freeQuantity?: number;
  productIds?: string[];
  code?: string;
  isActive: boolean;
  startsAt?: string;
  endsAt?: string;
  createdAt: string;
  updatedAt: string;
};

export type OfferInput = {
  id?: string;
  name: string;
  description: string;
  image?: string;
  type: OfferType;
  scope: OfferScope;
  percentage: number;
  fixedAmount: number;
  minPurchase: number;
  maxDiscount: number;
  buyQuantity: number;
  freeQuantity: number;
  productIds: string[];
  code: string;
  isActive: boolean;
  startsAt?: string | null;
  endsAt?: string | null;
};

export type OfferPreview = {
  subtotal: number;
  discountAmount: number;
  total: number;
  offerId?: string;
  offerName?: string;
  offerLabel?: string;
};

File: mizhara-backend/models/offer.go
package models

import "time"

type OfferType string
type OfferScope string

const (
	OfferTypePercentage OfferType = "percentage"
	OfferTypeFixed      OfferType = "fixed"
	OfferTypeBogo       OfferType = "bogo"
	OfferScopeAll       OfferScope = "all"
	OfferScopeSelected  OfferScope = "selected"
)

type Offer struct {
	ID           string     `json:"id"`
	Name         string     `json:"name"`
	Description  string     `json:"description"`
	Type         OfferType  `json:"type"`
	Scope        OfferScope `json:"scope"`
	Percentage   float64    `json:"percentage,omitempty"`
	FixedAmount  float64    `json:"fixedAmount,omitempty"`
	MinPurchase  float64    `json:"minPurchase,omitempty"`
	MaxDiscount  float64    `json:"maxDiscount,omitempty"`
	BuyQuantity  int        `json:"buyQuantity,omitempty"`
	FreeQuantity int        `json:"freeQuantity,omitempty"`
	ProductIDs   []string   `json:"productIds,omitempty"`
	Code         string     `json:"code,omitempty"`
	IsActive     *bool      `json:"isActive"`
	StartsAt     *time.Time `json:"startsAt,omitempty"`
	EndsAt       *time.Time `json:"endsAt,omitempty"`
	CreatedAt    time.Time  `json:"createdAt"`
	UpdatedAt    time.Time  `json:"updatedAt"`
}
